src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -0,0 +1,614 @@
+      |not be assigned to a sample given the criteria.  The output files will be the concatenation of sample id, sample
+      |name, and sample barcode bases (expected not observed), delimited by "-".  A metrics file will also be output
+      |providing analagous information to the metric desribed here:
+      |https://broadinstitute.github.io/picard/picard-metric-definitions.html#SampleBarcodeMetric
+      |
+      |FASTQs and associated read structures for each read structure should be given:
+      |- a single fragment read should have one FASTQ and one read structure
+      |- paired end reads should have two FASTQs and two read structures
+      |- a dual-index sample with paired end reads should have four FASTQs and four read structures given: two for the
+      |  two index reads, and two for the template reads.
+      |
+      |The read structures may contain sample barcode bases ('B'), molecular identifier bases ('M'), template bases ('T'),
+      |and bases to skip ('S'). Both reads must have teplate bases.  Any molecular identifiers will be concatenated using
+      |the '-' delimiter and placed in the given SAM record tag ("RX" by default).  Similarly, the sample barcode bases
+      |from the given read will be placed in the "BC" tag.
+      |
+      |The read group's sample id, sample name, and library id all correspond to the similarly named values in the
+      |sample sheet.  Library id will be the sample id if not found, and the platform unit will be the sample name
+      |concatenated with the sample barcode bases delimited by a ".".
+      |
+      |The sample section of the sample sheet should contain information related to each sample with the following
+      |columns
+      |  - Sample Identifier:  Sample_ID
+      |  - Sample Name:        Sample_Name
+      |  - Library Identifier: Library_ID
+      |  - Sample Project:     Sample_Project
+      |  - Description:        Description
+      |  - Sample Barcode:     Index
+      |The expected sample barcode should be placed in the 'Index' column.  If the sample barcode is present across
+      |multiple reads (ex. dual-index, or inline in both reads of a pair), then the expected barcode bases from each read
+      |should be concatenated and placed in teh 'Index' column.  The concatenation should be in the same order as the
+      |the order of the reads' FASTQs and read structures given to this tool.
+      |
+      |The read structures will be used to extract the observed sample barcode, template bses, and molecular identifiers
+      |from each read.  The observed sample barcode will be matched to the sample barcodes extracted from the bases in
+      |the sample sheet and associated read structures.
+    """,
+  group=ClpGroups.Fastq
+)
+class DemuxFastqs
+(@arg(flag="f", doc="One or more input fastq files each corresponding to a sub-read (ex. index read, read one, read two, fragment).") val fastqs:  Seq[PathToFastq],
```

Comment:
If you have multiple FASTQs, then yes, you will need to run this tool once per group of FASTQs, or at least combine them prior to running this tool.  As a by-product, the former allows for simple parallelization in a pipeline!